packages/web/utils/patronHelpers.ts

```diff
@@ -0,0 +1,59 @@
+import { Constants, Maybe } from '@metafam/utils';
+import { PlayerRank_Enum, TokenBalancesFragment } from 'graphql/autogen/types';
+
+export const PATRON_RANKS = [
+  PlayerRank_Enum.Diamond,
+  PlayerRank_Enum.Platinum,
+  PlayerRank_Enum.Gold,
+  PlayerRank_Enum.Silver,
+  PlayerRank_Enum.Bronze,
+];
+
+export const PATRONS_PER_RANK = [7, 7, 7, 14, 21];
+
+export const NUM_PATRONS = 150;
+export const MIN_PATRON_PSEEDS = 8;
+
+export const getLeagueCount = (rank: PlayerRank_Enum) => {
+  const index = PATRON_RANKS.findIndex((patronRank) => patronRank === rank);
+  return PATRONS_PER_RANK[index];
+};
+
+export const getLeagueCutoff = (
+  rank: PlayerRank_Enum,
+  pSeedHolders: TokenBalancesFragment[],
+  pSeedPrice: number,
+) => {
+  if (pSeedHolders == null || pSeedHolders.length === 0) {
+    return 0;
+  }
+  const leagueIndex = PATRON_RANKS.findIndex(
+    (patronRank) => patronRank === rank,
+  );
+  // this is the patron in the league with the least amount of holdings
+  const indexOfLastPatronInLeague =
+    PATRONS_PER_RANK.reduce(
+      (total, leagueCount, index) =>
+        leagueIndex < index ? total : total + leagueCount,
+      0,
+    ) - 1;
+  const pSeedCutoff =
+    pSeedHolders.length <= indexOfLastPatronInLeague
+      ? MIN_PATRON_PSEEDS
+      : parseFloat(pSeedHolders[indexOfLastPatronInLeague].pSeedBalance) /
+        Constants.PSEED_CONTRACT_DIVISOR;
+
+  return pSeedCutoff * pSeedPrice;
+};
+
+export const getPatronHoldingsUsd = (
+  pSeedHolder: Maybe<TokenBalancesFragment>,
+  pSeedPrice: number,
+) => {
+  if (pSeedHolder == null) {
+    return MIN_PATRON_PSEEDS * pSeedPrice;
+  }
+  const pSeedBalance =
```

Comment:
same here, if the balance is too large then this might break so we should use "BigNumber" math from `ethers`

Reply:
Ah, I figured at this point we have converted the 18+ decimal number down to 2+ decimals so we'd be good ?